### ISSUE TITLE
Powershell prompt replacement doesn't use existing prompt

### ISSUE THREAD

**Issue by ScottEvtuch** [conda/conda#8773]
## Current Behavior
Conda.psm1 renames Function:\prompt to Function:\CondaPromptBackup in the scope of function Add-CondaEnvironmentToPrompt when it should rename it in the global scope.

Opening a new PowerShell window after running "conda init" will load Conda.psm1 as part of the user's profile and the user will be presented with a prompt that contains the CONDA_PROMPT_MODIFIER environment variable followed by "PS >" without showing the current working directory.

### Steps to Reproduce

```
conda init
```

Open a new PowerShell window to load new profile

```
Windows PowerShell
Copyright (C) Microsoft Corporation. All rights reserved.

Loading personal and system profiles took 1515ms.
(base) PS>Get-Item Function:\CondaPromptBackup
Get-Item : Cannot find path 'Function:\CondaPromptBackup' because it does not exist.
At line:1 char:1
+ Get-Item Function:\CondaPromptBackup
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (Function:\CondaPromptBackup:String) [Get-Item], ItemNotFoundException
    + FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.GetItemCommand
```

## Expected Behavior
Conda.psm1 should rename the existing "prompt" function in the global scope so that it is accessible later to the new prompt function.

## Environment Information
<details open><summary><code>`conda info`</code></summary><p>

```
     active environment : base
    active env location : C:\tools\miniconda3
            shell level : 1
       user config file : C:\Users\sevtuch\.condarc
 populated config files :
          conda version : 4.6.14
    conda-build version : not installed
         python version : 3.7.3.final.0
       base environment : C:\tools\miniconda3  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/win-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/free/win-64
                          https://repo.anaconda.com/pkgs/free/noarch
                          https://repo.anaconda.com/pkgs/r/win-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/msys2/win-64
                          https://repo.anaconda.com/pkgs/msys2/noarch
          package cache : C:\tools\miniconda3\pkgs
                          C:\Users\sevtuch\.conda\pkgs
                          C:\Users\sevtuch\AppData\Local\conda\conda\pkgs
       envs directories : C:\tools\miniconda3\envs
                          C:\Users\sevtuch\.conda\envs
                          C:\Users\sevtuch\AppData\Local\conda\conda\envs
               platform : win-64
             user-agent : conda/4.6.14 requests/2.21.0 CPython/3.7.3 Windows/10 Windows/10.0.17763
          administrator : False
             netrc file : None
           offline mode : False
```
</p></details>

<details open><summary><code>`conda config --show-sources`</code></summary><p>

```

```
</p></details>

<details><summary><code>`conda list --show-channel-urls`</code></summary><p>

```
# packages in environment at C:\tools\miniconda3:
#
# Name                    Version                   Build  Channel
asn1crypto                0.24.0                   py37_0    defaults
astroid                   2.2.5                    py37_0    defaults
bcrypt                    3.1.6                    pypi_0    pypi
ca-certificates           2019.5.15                     0    defaults
certifi                   2019.3.9                 py37_0    defaults
cffi                      1.12.2           py37h7a1dbc1_1    defaults
chardet                   3.0.4                    py37_1    defaults
colorama                  0.4.1                    py37_0    defaults
conda                     4.6.14                   py37_0    defaults
console_shortcut          0.1.1                         3    defaults
cryptography              2.6.1            py37h7a1dbc1_0    defaults
idna                      2.8                      py37_0    defaults
isort                     4.3.20                   py37_0    defaults
jinja2                    2.10.1                   pypi_0    pypi
lazy-object-proxy         1.4.1            py37he774522_0    defaults
markupsafe                1.1.1                    pypi_0    pypi
mccabe                    0.6.1                    py37_1    defaults
menuinst                  1.4.16           py37he774522_0    defaults
ncclient                  0.6.6                    pypi_0    pypi
netaddr                   0.7.19                   pypi_0    pypi
openssl                   1.1.1c               he774522_1    defaults
paramiko                  2.5.0                    pypi_0    pypi
pip                       19.0.3                   py37_0    defaults
powershell_shortcut       0.0.1                         2    defaults
pycosat                   0.6.3            py37hfa6e2cd_0    defaults
pycparser                 2.19                     py37_0    defaults
pylint                    2.3.1                    py37_0    defaults
pynacl                    1.3.0                    pypi_0    pypi
pyopenssl                 19.0.0                   py37_0    defaults
pyserial                  3.4                      pypi_0    pypi
pysocks                   1.6.8                    py37_0    defaults
python                    3.7.3                h8c8aaf0_0    defaults
pywin32                   223              py37hfa6e2cd_1    defaults
pyyaml                    5.1.1                    pypi_0    pypi
requests                  2.21.0                   py37_0    defaults
ruamel_yaml               0.15.46          py37hfa6e2cd_0    defaults
scp                       0.13.2                   pypi_0    pypi
setuptools                41.0.0                   py37_0    defaults
six                       1.12.0                   py37_0    defaults
sqlite                    3.27.2               he774522_0    defaults
urllib3                   1.24.1                   py37_0    defaults
vc                        14.1                 h0510ff6_4    defaults
vs2015_runtime            14.15.26706          h3a45250_0    defaults
wheel                     0.33.1                   py37_0    defaults
win_inet_pton             1.1.0                    py37_0    defaults
wincertstore              0.2                      py37_0    defaults
wrapt                     1.11.1           py37he774522_0    defaults
yaml                      0.1.7                hc54c509_2    defaults
```
</p></details>

**Comment by Chaoste** [comment]
For me I was able to fix this by writing the following code after the initialization part of conda in my powershell profile:

```
function prompt {
  $conda_env = conda env list | grep '*' | sed 's/ .*//'
  $p = Get-Location
  "$p ($conda_env)> "
}
```
I am using common UNIX commands like grep and sed since they are installed on my system via clink (I guess). But if this is not available one might replace this with appropiate Powershell commands. For now you may also leave the environment part and just show the path.

**Comment by dennispg** [comment]
I wish I had seen this sooner, but I don't think the fix for this was the right path forward.

The existing prompt function getting overwritten was moved within the scope of `Add-CondaEnvironmentToPrompt` because there are times when you do not want that to happen.

Specifically when changeps1 is set to false, it should not touch the prompt at all. Please see #8465

**Comment by ScottEvtuch** [comment]
So from a PowerShell architecture point of view, the task we're trying to accomplish _can't_ happen in the scope of a function otherwise it won't have proper access to the global variables. There may be a way to work around this but I don't know that it is possible. Essentially we cannot rename functions in the global scope from within a function.

I wonder if replacing "Rename-Item" with "Copy-Item" would result in the desired behavior. That way we have a good backup copy of the user's custom prompt _if_ they want to run "Add-CondaEnvironmentToPrompt" but we don't blow away their existing prompt if they don't run it.

**Comment by dennispg** [comment]
I think that makes the most sense, take a copy of the prompt instead of renaming. Should the copy then happen back inside the scope of `Add-CondaEnvironmentToPrompt` or outside like it is right now?

For my own better understanding, what exactly is it about the access to the global variables that wasn't working? `Add-CondaEnvironmentToPrompt` is able to overwrite `prompt()` in the global scope from within the function by defining it directly instead of using `Rename-Item`. Is it just that `Rename-Item` can't operate against the global scope?

**Comment by ScottEvtuch** [comment]
You've got it exactly right. Rename-Item can't operate on functions in the global scope, as far as I've been able to figure out. It only renames functions in the current scope. I'm assuming Copy-Item also functions in the same way.

For a while I played around with the idea of saving the definition of the current global function to a temporary variable and then creating a new global one with that, but it wasn't nearly as elegant and reliable as using "Rename-Item"
